Fix #20694: Fixed search bar moving when using search

TheSolarz opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #20694 
2. This PR does the following: 
    1. Aligns the search bar and its results to the top of the page in the /community-library tab by using the `mx-auto` CSS class instead of `m-auto`. 
    2. Removed extra (scrollable) space at the bottom of the page when few search results are present caused by the  `min-height: 100vh` on the `oppia-library-container`. This has been solved by creating a parent wrapper div for the `classroom-cards-wrapper` div (which is not always present) and the `oppia-library-container` div. This parent div controls the background color and has a reduced `min-height: calc(100vh - 56px)` compared to the previous child-div `oppia-library-container` having a `min-height: 100vh`. The reduced `56px` is from the `navbar-container`'s height (which is consistently `56px`).
3. The original bug occurred because: 
    1. Incorrect use of `m-auto` instead of `mx-auto` in PR #14991 
    2. In the same PR #14991 the incorrect `min-height: 100vh` on the `oppia-library-container` div is introduced

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [x] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [ ] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [x] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).

## Proof that changes are correct

<!--
Add before-and-after videos/screenshots of the user-facing interface (including
the browser devtools console) to demonstrate that the changes made in this PR
work correctly. Make sure the actions taken in the before and after videos are
the same. (For changes involving responsiveness or adjustment of UI elements,
this should be a video that gradually resizes the viewport from wide to narrow
and back again.) If this PR is for a developer-facing feature, provide
videos/screenshots of the developer-facing interface instead.

When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->

Before:
https://github.com/user-attachments/assets/af7e2d95-6632-4a03-a6cc-6c6f5a215058

After (note that this shows both with and without a dummy math classroom):
https://github.com/user-attachments/assets/5d1879f3-fb0d-4869-a696-f2bdd4776082

#### Proof of changes on desktop with slow/throttled network
No weird behavior observed.

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone
See video proof.

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language
No observed changes when the language is set to Arabic.

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

TheSolarz committed:
Changed m-auto to mx-auto to align search and search results to the top-middle

TheSolarz committed:
Reduced min-height by 400px for oppia-library-container to redcuce blank space for a search with few results

TheSolarz committed:
Removed min-height requirement for when page width is less than 900px

TheSolarz committed:
Added a parent wrapper div that controls the background-color instead of working around the optional classroom-cards-wrapper div

TheSolarz committed:
Merge branch 'oppia:develop' into search-bar-moving-when-typing

TheSolarz committed:
Merge branch 'oppia:develop' into search-bar-moving-when-typing

Nik-09 reviewed (changes requested):
Hi @TheSolarz some tests are failing can you please fix these tests?
Feel free to reassign me once done.

Thanks

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Fix linting issues with trailing whitespace

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Fix linting issues after conflict merge

seanlip committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Removed mx-auto as that was causing tests to fail

This was thought to be a flake and had previously been reported as one as well (#20855), but it turned out to be mx-auto causing tests to fail by not having Explorations/Collections be clickable

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

TheSolarz committed:
Merge branch 'develop' into search-bar-moving-when-typing

seanlip commented:
I think @Nik-09 needs to review this actually -- @TheSolarz, FYI, if you don't assign a reviewer then they won't see your message. To assign, please leave a comment along the lines of "@reviewer PTAL". Thanks!

Nik-09 committed:
Merge branch 'develop' into search-bar-moving-when-typing

Nik-09 commented:
I have updated the branch, let's see the current status of the CI checks.

Nik-09 reviewed (approved):
LGTM, thank you @TheSolarz

Muskan-2005-coder commented:
can you please assign me this issue?
